Index: oxide-qt-1.2.0/CMakeLists.txt
===================================================================
--- oxide-qt-1.2.0.orig/CMakeLists.txt	2014-09-03 16:08:28.077973471 +0100
+++ oxide-qt-1.2.0/CMakeLists.txt	2014-09-03 16:08:28.065973471 +0100
@@ -101,7 +101,7 @@
   list(APPEND GYP_COMMAND -Dproprietary_codecs=1)
 else()
   list(APPEND GYP_COMMAND -Dffmpeg_branding=Chromium)
-  list(APPEND GYP_COMMAND -Dproprietary_codecs=0)
+  list(APPEND GYP_COMMAND -Dproprietary_codecs=1)
 endif()
 if(DEFINED ENABLE_PLUGINS AND ENABLE_PLUGINS EQUAL 0)
   list(APPEND GYP_COMMAND -Denable_plugins=0)
Index: oxide-qt-1.2.0/third_party/chromium/src/content/renderer/renderer_webkitplatformsupport_impl.cc
===================================================================
--- oxide-qt-1.2.0.orig/third_party/chromium/src/content/renderer/renderer_webkitplatformsupport_impl.cc	2014-09-03 16:08:28.077973471 +0100
+++ oxide-qt-1.2.0/third_party/chromium/src/content/renderer/renderer_webkitplatformsupport_impl.cc	2014-09-03 16:08:28.065973471 +0100
@@ -121,6 +121,12 @@
 #include "content/renderer/media/webrtc/peer_connection_dependency_factory.h"
 #endif
 
+#if defined(USE_PROPRIETARY_CODECS)
+#include "media/ffmpeg/ffmpeg_common.h"
+#include "media/filters/ffmpeg_glue.h"
+#include "third_party/ffmpeg/ffmpeg_stubs.h"
+#endif
+
 using blink::Platform;
 using blink::WebAudioDevice;
 using blink::WebBlobRegistry;
@@ -403,6 +409,15 @@
   if (!net::IsSupportedMediaMimeType(mime_type_ascii))
     return IsNotSupported;
 
+#if defined(USE_PROPRIETARY_CODECS)
+  media::FFmpegGlue::InitializeFFmpeg();
+  if (net::IsProprietaryMediaMimeType(mime_type_ascii) &&
+      !avcodec_find_decoder(AV_CODEC_ID_H264)) {
+    DVLOG(1) << "Proprietary codecs are not installed";
+    return IsNotSupported;
+  }
+#endif
+
   if (!key_system.isEmpty()) {
     // Check whether the key system is supported with the mime_type and codecs.
 
@@ -451,6 +466,16 @@
   net::ParseCodecString(ToASCIIOrEmpty(codecs), &parsed_codec_ids, false);
   if (mime_type_ascii.empty())
     return false;
+
+#if defined(USE_PROPRIETARY_CODECS)
+  media::FFmpegGlue::InitializeFFmpeg();
+  if (net::IsProprietaryMediaMimeType(mime_type_ascii) &&
+      !avcodec_find_decoder(AV_CODEC_ID_H264)) {
+    DVLOG(1) << "Proprietary codecs are not installed";
+    return false;
+  }
+#endif
+
   return media::StreamParserFactory::IsTypeSupported(
       mime_type_ascii, parsed_codec_ids);
 }
Index: oxide-qt-1.2.0/third_party/chromium/src/net/base/mime_util.cc
===================================================================
--- oxide-qt-1.2.0.orig/third_party/chromium/src/net/base/mime_util.cc	2014-09-03 16:08:28.077973471 +0100
+++ oxide-qt-1.2.0/third_party/chromium/src/net/base/mime_util.cc	2014-09-03 16:08:28.065973471 +0100
@@ -1340,6 +1340,15 @@
   return file_type != CERTIFICATE_MIME_TYPE_UNKNOWN;
 }
 
+bool IsProprietaryMediaMimeType(const std::string& mime_type) {
+  for (size_t i = 0; i < arraysize(proprietary_media_types); ++i) {
+    if (mime_type == proprietary_media_types[i]) {
+      return true;
+    }
+  }
+  return false;
+}
+
 void AddMultipartValueForUpload(const std::string& value_name,
                                 const std::string& value,
                                 const std::string& mime_boundary,
Index: oxide-qt-1.2.0/third_party/chromium/src/net/base/mime_util.h
===================================================================
--- oxide-qt-1.2.0.orig/third_party/chromium/src/net/base/mime_util.h	2014-09-03 16:08:28.077973471 +0100
+++ oxide-qt-1.2.0/third_party/chromium/src/net/base/mime_util.h	2014-09-03 16:08:28.065973471 +0100
@@ -46,6 +46,8 @@
 NET_EXPORT bool IsSupportedJavascriptMimeType(const std::string& mime_type);
 NET_EXPORT bool IsSupportedCertificateMimeType(const std::string& mime_type);
 
+NET_EXPORT bool IsProprietaryMediaMimeType(const std::string& mime_type);
+
 // Convenience function.
 NET_EXPORT bool IsSupportedMimeType(const std::string& mime_type);
 
Index: oxide-qt-1.2.0/third_party/chromium/src/content/content_renderer.gypi
===================================================================
--- oxide-qt-1.2.0.orig/third_party/chromium/src/content/content_renderer.gypi	2014-09-03 16:08:28.077973471 +0100
+++ oxide-qt-1.2.0/third_party/chromium/src/content/content_renderer.gypi	2014-09-03 16:08:28.069973471 +0100
@@ -17,6 +17,7 @@
     '../mojo/mojo_base.gyp:mojo_js_bindings_lib',
     '../net/net.gyp:net',
     '../skia/skia.gyp:skia',
+    '../third_party/ffmpeg/ffmpeg.gyp:ffmpeg',
     '../third_party/WebKit/public/blink.gyp:blink',
     '../third_party/icu/icu.gyp:icui18n',
     '../third_party/icu/icu.gyp:icuuc',
