Index: oxide-qt-1.21.0~git1483696852.71548c4/oxide/shared/renderer/oxide_content_renderer_client.cc
===================================================================
--- oxide-qt-1.21.0~git1483696852.71548c4.orig/oxide/shared/renderer/oxide_content_renderer_client.cc
+++ oxide-qt-1.21.0~git1483696852.71548c4/oxide/shared/renderer/oxide_content_renderer_client.cc
@@ -54,6 +54,12 @@
 #include "media/oxide_web_media_player.h"
 #endif
 
+#if defined(USE_PROPRIETARY_CODECS)
+#include "media/base/mime_util.h"
+#include "media/ffmpeg/ffmpeg_common.h"
+#include "media/filters/ffmpeg_glue.h"
+#endif
+
 namespace oxide {
 
 void ContentRendererClient::RenderThreadStarted() {
@@ -74,6 +80,14 @@ void ContentRendererClient::RenderThread
   // Oxide does not support NavigatorContentUtils.
   // See https://launchpad.net/bugs/1214046
   blink::WebRuntimeFeatures::enableNavigatorContentUtils(false);
+
+#if defined(USE_PROPRIETARY_CODECS)
+  media::FFmpegGlue::InitializeFFmpeg();
+  if (!avcodec_find_decoder(AV_CODEC_ID_H264)) {
+    DVLOG(1) << "Proprietary codecs are not installed";
+    media::RemoveProprietaryMediaTypesAndCodecsForTests();
+  }
+#endif
 }
 
 void ContentRendererClient::RenderFrameCreated(
Index: oxide-qt-1.21.0~git1483696852.71548c4/oxide/build/cmake/ChromiumBuildShim.cmake
===================================================================
--- oxide-qt-1.21.0~git1483696852.71548c4.orig/oxide/build/cmake/ChromiumBuildShim.cmake
+++ oxide-qt-1.21.0~git1483696852.71548c4/oxide/build/cmake/ChromiumBuildShim.cmake
@@ -210,7 +210,7 @@ function(run_generate_ninja)
     list(APPEND MAKE_GN_ARGS_CMD -Dproprietary_codecs=true)
   else()
     list(APPEND MAKE_GN_ARGS_CMD -Dffmpeg_branding=Chromium)
-    list(APPEND MAKE_GN_ARGS_CMD -Dproprietary_codecs=false)
+    list(APPEND MAKE_GN_ARGS_CMD -Dproprietary_codecs=true)
   endif()
 
   if(ENABLE_PLUGINS)
Index: oxide-qt-1.21.0~git1483696852.71548c4/media/BUILD.gn
===================================================================
--- oxide-qt-1.21.0~git1483696852.71548c4.orig/media/BUILD.gn
+++ oxide-qt-1.21.0~git1483696852.71548c4/media/BUILD.gn
@@ -28,12 +28,6 @@ buildflag_header("media_features") {
   ]
 }
 
-if (proprietary_codecs && media_use_ffmpeg) {
-  assert(
-      ffmpeg_branding != "Chromium",
-      "proprietary codecs and ffmpeg_branding set to Chromium are incompatible")
-}
-
 # Common configuration for targets in the media directory.
 # NOT for exporting.
 config("media_config") {
Index: oxide-qt-1.21.0~git1483696852.71548c4/oxide/shared/BUILD.gn
===================================================================
--- oxide-qt-1.21.0~git1483696852.71548c4.orig/oxide/shared/BUILD.gn
+++ oxide-qt-1.21.0~git1483696852.71548c4/oxide/shared/BUILD.gn
@@ -216,6 +216,7 @@ component("shared") {
     "//services/ui/public/cpp/gpu",
     "//skia",
     "//storage/common",
+    "//third_party/ffmpeg",
     "//third_party/WebKit/public:blink",
     "//ui/base",
     "//ui/base/ime:text_input_types",
