Index: oxide-qt-1.14.0~bzr1366/CMakeLists.txt
===================================================================
--- oxide-qt-1.14.0~bzr1366.orig/CMakeLists.txt	2016-03-03 11:02:44.007915273 +0000
+++ oxide-qt-1.14.0~bzr1366/CMakeLists.txt	2016-03-03 11:02:43.955914949 +0000
@@ -255,7 +255,7 @@
   list(APPEND _GYP_COMMAND -Dproprietary_codecs=1)
 else()
   list(APPEND _GYP_COMMAND -Dffmpeg_branding=Chromium)
-  list(APPEND _GYP_COMMAND -Dproprietary_codecs=0)
+  list(APPEND _GYP_COMMAND -Dproprietary_codecs=1)
 endif()
 if(ENABLE_PLUGINS)
   list(APPEND _GYP_COMMAND -Denable_plugins=1)
Index: oxide-qt-1.14.0~bzr1366/shared/renderer/oxide_content_renderer_client.cc
===================================================================
--- oxide-qt-1.14.0~bzr1366.orig/shared/renderer/oxide_content_renderer_client.cc	2016-03-03 11:02:44.007915273 +0000
+++ oxide-qt-1.14.0~bzr1366/shared/renderer/oxide_content_renderer_client.cc	2016-03-03 11:02:43.955914949 +0000
@@ -55,6 +55,12 @@
 #include "media/oxide_web_media_player.h"
 #endif
 
+#if defined(USE_PROPRIETARY_CODECS)
+#include "media/base/mime_util.h"
+#include "media/ffmpeg/ffmpeg_common.h"
+#include "media/filters/ffmpeg_glue.h"
+#endif
+
 namespace oxide {
 
 void ContentRendererClient::RenderThreadStarted() {
@@ -75,6 +81,14 @@
   // Oxide does not support NavigatorContentUtils.
   // See https://launchpad.net/bugs/1214046
   blink::WebRuntimeFeatures::enableNavigatorContentUtils(false);
+
+#if defined(USE_PROPRIETARY_CODECS)
+  media::FFmpegGlue::InitializeFFmpeg();
+  if (!avcodec_find_decoder(AV_CODEC_ID_H264)) {
+    DVLOG(1) << "Proprietary codecs are not installed";
+    media::RemoveProprietaryMediaTypesAndCodecsForTests();
+  }
+#endif
 }
 
 void ContentRendererClient::RenderFrameCreated(
Index: oxide-qt-1.14.0~bzr1366/shared/shared.gyp
===================================================================
--- oxide-qt-1.14.0~bzr1366.orig/shared/shared.gyp	2016-03-03 11:02:44.007915273 +0000
+++ oxide-qt-1.14.0~bzr1366/shared/shared.gyp	2016-03-03 11:02:43.955914949 +0000
@@ -250,6 +250,7 @@
         '<(DEPTH)/mojo/mojo_public.gyp:mojo_cpp_bindings',
         '<(DEPTH)/net/net.gyp:net',
         '<(DEPTH)/skia/skia.gyp:skia',
+        '<(DEPTH)/third_party/ffmpeg/ffmpeg.gyp:ffmpeg',
         '<(DEPTH)/third_party/khronos/khronos.gyp:khronos_headers',
         '<(DEPTH)/third_party/libXNVCtrl/libXNVCtrl.gyp:libXNVCtrl',
         '<(DEPTH)/third_party/re2/re2.gyp:re2',
