Description: fix pointer alignment in V8 on arm
Bug: https://bugs.chromium.org/p/v8/issues/detail?id=5668
Author: Chris Coulson <chris.coulson@canonical.com>

Index: oxide-qt-1.19.5/v8/src/zone/zone.cc
===================================================================
--- oxide-qt-1.19.5.orig/v8/src/zone/zone.cc
+++ oxide-qt-1.19.5/v8/src/zone/zone.cc
@@ -87,6 +87,9 @@
 
   // Check that the result has the proper alignment and return it.
   DCHECK(IsAddressAligned(result, kAlignment, 0));
+  if (kPointerSize == 4 && kAlignment == 4) {
+    DCHECK((size & 4) || IsAddressAligned(result, 8, 0));
+  }
   allocation_size_ += size;
   return reinterpret_cast<void*>(result);
 }
@@ -209,6 +212,9 @@
 
   // Recompute 'top' and 'limit' based on the new segment.
   Address result = RoundUp(segment->start(), kAlignment);
+  if (kPointerSize == 4 && kAlignment == 4) {
+    result += ((~size) & 4) & (reinterpret_cast<intptr_t>(result) & 4);
+  }
   position_ = result + size;
   // Check for address overflow.
   // (Should not happen since the segment is guaranteed to accomodate
